### PR TITLE
Clear some shadow warnings

### PR DESCRIPTION
**Pull request by starseeker**
Ran across a few -Wshadow warnings when building this code.  This change just attempts to clear them by renaming, and commenting out one unused var in DockContainerWidget.cpp.

### CODE REVIEW

**Comment by githubuser0xFFFF** [comment]
Thank you. May I ask you what compiler do you use?

**Comment by starseeker** [comment]
In this case it was gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2)

**Comment by githubuser0xFFFF** [comment]
Could you please post the compiler output to help me get a better understanding. Sorry for the additional work but I have to admit, I don't like this change.

Writing code means writing understandable, clean, readable and maintainable code for other people. This includes a good naming for variables. Prefixing local variable names with a big L makes the code harder to read and is not for humans but just to silence the compiler.

This is my feeling. Therefore I would like to understand the problem of the compiler. 

Thank you

**Comment by starseeker** [comment]
No problem - I wasn't sure my proposed fix would be the preferred one; the main idea was to identify that the issue was there.  Below is a sample output (-Wshadow and -Werror flags are added to the build):

[  4%] Building CXX object src/CMakeFiles/qtadvanceddocking.dir/DockAreaTitleBar.cpp.o
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp: In member function ‘void ads::DockAreaTitleBarPrivate::createButtons()’:
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:152:9: error: declaration of ‘TabsMenu’ shadows a member of ‘ads::DockAreaTitleBarPrivate’ [-Werror=shadow]
  152 |  QMenu* TabsMenu = new QMenu(TabsMenuButton);
      |         ^~~~~~~~
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:73:9: note: shadowed declaration is here
   73 |  QMenu* TabsMenu;
      |         ^~~~~~~~
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp: In member function ‘ads::IFloatingWidget* ads::DockAreaTitleBarPrivate::makeAreaFloating(const QPoint&, ads::eDragState)’:
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:212:102: error: declaration of ‘DragState’ shadows a member of ‘ads::DockAreaTitleBarPrivate’ [-Werror=shadow]
  212 | IFloatingWidget* DockAreaTitleBarPrivate::makeAreaFloating(const QPoint& Offset, eDragState DragState)
      |                                                                                                      ^
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:77:13: note: shadowed declaration is here
   77 |  eDragState DragState = DraggingInactive;
      |             ^~~~~~~~~
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:219:19: error: declaration of ‘FloatingWidget’ shadows a member of ‘ads::DockAreaTitleBarPrivate’ [-Werror=shadow]
  219 |  IFloatingWidget* FloatingWidget;
      |                   ^~~~~~~~~~~~~~
/home/user/Qt-Advanced-Docking-System/src/DockAreaTitleBar.cpp:78:19: note: shadowed declaration is here
   78 |  IFloatingWidget* FloatingWidget = nullptr;
      |                   ^~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/qtadvanceddocking.dir/build.make:148: src/CMakeFiles/qtadvanceddocking.dir/DockAreaTitleBar.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:218: src/CMakeFiles/qtadvanceddocking.dir/all] Error 2
make: *** [Makefile:147: all] Error 2

**Comment by rolivav** [comment]
Usually code bases tend to name differently member variables to both see clearly when a variable is a member variable and make it less likely to encounter these issues with shadowing variables. In the Qt code base they tend to use `m_` as a prefix. I usually use `m`, I feel it fits better with the lower camel case style.

**Comment by githubuser0xFFFF** [comment]
@rolivav Yes I agree. But normally the private members are already prefixed with the d member pointer:

```c++
d->Value1 = 1;
d->Value2 = 5;
```

The problem is, if the private members are accessed from private functions. If I use the m_ prefix that I usually use for private members, then it would look like this if I use the members from functions in the public interface class:

```
d->m_Value1 = 1;
d->m_Value2 = 5;
```

That is quite ugly. But I think this will be the solution. It is at least better then to prefix al local variables with an L.

**Comment by rolivav** [comment]
Indeed @githubuser0xFFFF , that is one of the many weird legibility problems that pimpls create and why I try not to use them. They do have their benefits, especially in a lib like this, I'm not judging :) .
>That is quite ugly. But I think this will be the solution. But I would prefer this more then to prefix al local variables with an L.
I agree, but the prefix `d->` is quite small and doesn't bloat the code too much. It is also very descriptive: it is telling me this is a member variable that is inside the pimpl, I would not confuse it with anything else and have to lose time checking it.
```
d->m_Value1 = 1; // inside pimpl
m_Value1 = 1; // defined in header
```
These little things improve a lot code navigation when they add up.

File: src/DockManager.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ enum eStateFileVersion
 	CurrentVersion = Version1//!< CurrentVersion
 };
 
-static CDockManager::ConfigFlags StaticConfigFlags = CDockManager::DefaultNonOpaqueConfig;
+static CDockManager::m_ConfigFlags StaticConfigFlags = CDockManager::DefaultNonOpaqueConfig;
```
Comment (rolivav) [inline src/DockManager.cpp:91]: Class definitions should not have the `m_` prefix.

Comment (starseeker) [reply src/DockManager.cpp:91]: Not a problem - what would you suggest to avoid the ConfigFlags conflict there?

**Comment by githubuser0xFFFF** [comment]
How about disable -Wshadow?

**Comment by rolivav** [comment]
@githubuser0xFFFF in my opinion libs should aim to build without any warnings. It would be amazing to change the warning level of this project to treat warnings as errors and fix all of them, really helps with maintainability.

File: demo/MainWindow.cpp [hunk 2]
```diff
@@ -446,10 +446,10 @@ void MainWindowPrivate::createContent()
     }
 #endif
 
-	for (auto DockWidget : DockManager->dockWidgetsMap())
+	for (auto m_DockWidget : DockManager->dockWidgetsMap())
```
Comment (rolivav) [inline demo/MainWindow.cpp:449]: This is a local variable.

File: src/DockManager.cpp (hunk 1 above)
Comment (rolivav) [reply src/DockManager.cpp:91]: Where is the conflict?

File: demo/MainWindow.cpp [hunk 3]
```diff
@@ -419,10 +419,10 @@ void MainWindowPrivate::createContent()
...
 	QObject::connect(CustomButton, &QToolButton::clicked, [=]()
 	{
-		auto DockWidget = createEditorWidget();
-		DockWidget->setFeature(ads::CDockWidget::DockWidgetDeleteOnClose, true);
-		DockManager->addDockWidgetTabToArea(DockWidget, DockArea);
-		_this->connect(DockWidget, SIGNAL(closeRequested()), SLOT(onEditorCloseRequested()));
+		auto clicked_DockWidget = createEditorWidget();
+		clicked_DockWidget->setFeature(ads::CDockWidget::DockWidgetDeleteOnClose, true);
+		DockManager->addDockWidgetTabToArea(clicked_DockWidget, DockArea);
+		_this->connect(clicked_DockWidget, SIGNAL(closeRequested()), SLOT(onEditorCloseRequested()));
```
Comment (rolivav) [inline demo/MainWindow.cpp:425]: I would leave the variable as it was, it doesn't match the coding standard of this repo.

**Comment by rolivav** [comment]
@starseeker I would ask @githubuser0xFFFF if he plans to merge this at all. In my opinion this change would be better to make it to the whole repo instead of having it split in pieces.
I would suggest that you integrate this lib in a different project where you can disable all warnings, as I do.

**Comment by starseeker** [comment]
Doubt we can find a change set that will be a good fit without diminishing returns, closing request - thanks to those who took the time to look.

**Pull request closed without merging**